samples/tools/autogenbench/scenarios/WebArena/Scripts/init_tasks.py

```diff
@@ -89,12 +90,19 @@ def main():
         if entry.is_dir():
             templates[re.sub(r"\s", "", entry.name)] = entry.path
 
-    # Divide the tasks by their websites
+    # Divide the tasks by their websites and if they are validation or test
     page_groups = dict()
     for task in tasks:
+
+        # We don't know how the intent id's are distributed, so hash them to get a uniform dist
+        template_hash = hashlib.md5(str(task["intent_template_id"]).encode("utf-8")).hexdigest()
+        task_set = "validation" if template_hash[0] in "01234567" else "test"
```

Comment:
I don't understand this line. Can you add comments?

Reply:
Have a look at new comments. Let me know if they are sufficient. Recopied here for convenience:

```
        # The full hash will consist of 32 hexadecimal digits. We can get a 50/50 split by checking if the first digit is in the range (0-7) vs (8-F)

```